articles/advanced-r-concepts/index.md

```diff
@@ -0,0 +1,333 @@
+
+        1. n: Number of samples
+        2. min: lower bound of the range
+        3. max:  upper bound of the range
+
+2. **For loops**: For demonstrating for loops, we will consider printing the elements of a list and the matrix. Since a matrix is a 2-D data structure, we will need two loops. Such usage of a loop within a loop is referred to as nested looping.
+
+    ```r
+    for(i in 1:10){
+        print(i)
+    }
+    # double for loop
+    x<- matrix[1:6, 2:3]
+    x
+    for(i in seq_len(nrow(x))){
+        for(j in seq_len(ncol(x))){
+            print(x[i,j])
+        }
+    }
+    ```
+
+Notice the new `seq_len` function. We have just learnt a new function. This is similar to the `range` function in Python. `seq_len(x)` generates integers from 1 to x. Note that R is 1-indexed langugae. All indices start with one. The output for the above code output is as follows:
+
+        ```txt
+        > x
+            [,1] [,2] [,3]
+        [1,]    1    3    5
+        [2,]    2    4    6
+
+        <!-- for loop output -->
+
+        [1] 1
+        [1] 3
+        [1] 5
+        [1] 2
+        [1] 4
+        [1] 6
+        ```
+3. **While loops**: While loops are known for their conditional execution of loops. They lack the iterator variable that the for loop uses. Let us look at the syntax of while loops in R:
+
+    ```txt
+    while (condition_expression)
+    {
+    execute_code_while_condition_is_true
+    }
+    ```
+    Consider the following example:
+    ```r
+    iter <-10
+    while (iter>3){
+        print(iter)
+        iter<iter-1
+    }
+    ```
+    The output of the above is as follows:
+
+    ```txt
+    10
+    9
+    8
+    7
+    6
+    5
+    4
+    ```
+When the variable `iter` equals three, the condition fails. Therefore, the print statement stops executing.
+
+### Functions
+Functions are important aspects of all programming languages. They help modularize the code. They help implement various programming paradigms such as Don't Repeat Yourself- [DRY](https://www.section.io/engineering-education/dry-manifesto) and modularity.  
+
+Let us look at function definition as following in R:
+
+```r
+function_name <- function(parameter_1=10){
+    hello <- "Hellooooo"
+    for i in seq_len(parameter_1){
+        print(hello)
+    }
+    return (parameter_1 + 10)
+}
+```
+To call the function, we pass the parameter to the function name directly:
+```r
+function_name(10)
+```
+
+In cases when multiple values need to be returned, we use lists. This way, a single entity with various outputs is being returned. I suggest you try this last point out as a challenge.
+
+### Conclusion
+In this article, we have dealt with the advanced topics of the R programming language. At this point, you have the skills to master machine learning algorithms and perform various data science experiments. Adios.
```

Comment:
Remove Adios.

I would not go so far as to say you can master machine learning algorithms, but you can say at this point you know the main functions and how to use them.